commands/incr.md

```diff
@@ -69,25 +69,32 @@ The more simple and direct implementation of this pattern is the following:
 FUNCTION LIMIT_API_CALL(ip)
 ts = CURRENT_UNIX_TIME()
 keyname = ip+":"+ts
+WATCH(keyname)
 current = GET(keyname)
 IF current != NULL AND current > 10 THEN
     ERROR "too many requests per second"
 ELSE
     MULTI
         INCR(keyname,1)
         EXPIRE(keyname,10)
-    EXEC
-    PERFORM_API_CALL()
+    reply = EXEC
+    IF reply != NULL
+        PERFORM_API_CALL()
+    ELSE
+      RETURN LIMIT_API_CALL(ip)
```

Comment:
I wrote this differently: https://github.com/antirez/redis-doc/pull/893/files#diff-70e66b9c0c88bb9f371d12bb2bb9e4d8R112

It is more clear what the intent is. Making the recursive call directly is an implementation detail